### PR TITLE
Simplify the OSS documentation deploy system (#2670)

### PR DESCRIPTION
**Pull request by tkorenko**
## What
+ Adds Slack notification about failed Docs builds
+ (while here) Unbreaks docs build

### CODE REVIEW

**Review by supertopher — APPROVED** [review]
Looks great Taras.  thank you for adding the slack add-on I think it will be really helpful to the team

File: .github/workflows/deploy-docs-site.yml [hunk 1]
```diff
@@ -13,25 +13,39 @@ on:
...
           cache-dependency-path: docusaurus
 
       - name: Run Docusaurus
-        if: github.event_name == 'push' || github.event_name == 'workflow_dispatch'
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: ./tools/bin/deploy_docusaurus
+
+      - name: Notify Slack channel on failure
+        uses: abinoda/slack-action@master
```
Comment (cuyk) [inline .github/workflows/deploy-docs-site.yml:38]: Would you mind switching to the official slackapi action?
https://github.com/slackapi/slack-github-action

Comment (cuyk) [reply .github/workflows/deploy-docs-site.yml:38]: Already merged, nevermind =)

Comment (tkorenko) [reply .github/workflows/deploy-docs-site.yml:38]: There's more workflows to switch to the official API, which means we might want to start a separate issue for OSS repo. Or both repos.

**Pull request merged**